packages/flutter_tools/lib/src/usage.dart

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:usage/usage_io.dart';
+
+import 'base/context.dart';
+import 'globals.dart';
+import 'runner/version.dart';
+
+// TODO(devoncarew): We'll need to do some work on the user agent in order to
+// correctly track usage by operating system (dart-lang/usage/issues/70).
+
+// TODO(devoncarew): We'll want to find a way to send (sanitized) command parameters.
+
+const String _kFlutterUA = 'UA-67589403-5';
+
+const Duration _kMaxExitDelayDuration = const Duration(milliseconds: 250);
+
+class Usage {
+  /// Returns [Usage] active in the current app context.
+  static Usage get instance => context[Usage] ?? (context[Usage] = new Usage());
+
+  Usage() {
+    _ga = new AnalyticsIO(_kFlutterUA, 'flutter', FlutterVersion.getVersionString());
+
+    // Check if this is the first run. If so, enable analytics. We show opt-out
+    // methods (flutter config) the first time the tool is run.
+    if (!_ga.hasSetOptIn) {
+      _isFirstRun = true;
```

Comment:
I meant on the first run. The one where we say how to opt-out. But yeah, not tracking the config command seems like an excellent idea also!